### PR TITLE
Extend python API for GenerateProposals

### PR DESCRIPTION
**Pull request by meiyang-intel**
### Details:
 - *Extend nGraph python API for GenerateProposals*

### Tickets:
 - *CVS-79284*

### CODE REVIEW

File: src/bindings/python/src/openvino/runtime/opset9/ops.py [hunk 1]
```diff
@@ -104,3 +104,45 @@ def softsign(node: NodeInput, name: Optional[str] = None) -> Node:
     :return: New node with SoftSign operation applied on each element of it.
     """
     return _get_node_factory_opset9().create("SoftSign", [as_node(node)], {})
+
+@nameable_op
```
Comment (jiwaszki) [inline src/bindings/python/src/openvino/runtime/opset9/ops.py:108]:
```suggestion


@nameable_op
```

**Review by jiwaszki — CHANGES_REQUESTED** [review]
The operator must be added to `openvino/src/bindings/python/src/compatibility/ngraph/opset9/` as well.

File: src/bindings/python/src/openvino/runtime/opset9/ops.py [hunk 2]
```diff
@@ -146,3 +146,49 @@ def irdft(
...
+def generate_proposals(
+    im_info: NodeInput,
+    anchors: NodeInput,
+    deltas: NodeInput,
+    scores: NodeInput,
+    min_size: float,
+    nms_threshold: float,
+    post_nms_count: int,
+    pre_nms_count: int,
```
Comment (jiwaszki) [inline src/bindings/python/src/openvino/runtime/opset9/ops.py:160]: Please follow docs when setting attributes order: https://github.com/openvinotoolkit/openvino/blob/master/docs/ops/detection/GenerateProposals_9.md

@meiyang-intel @luo-cheng2021 I tracked C++ implementation back to you. Could you update it to follow docs order of attributes so PyAPI can fallback to it without any issues?

```suggestion
    pre_nms_count: int,
    post_nms_count: int,
```

File: src/bindings/python/src/openvino/runtime/opset9/ops.py [hunk 3]
```diff
@@ -146,3 +146,49 @@ def irdft(
...
+    inputs = as_nodes(im_info, anchors, deltas, scores)
+
+    attributes = {
+        "min_size": min_size,
+        "nms_threshold": nms_threshold,
+        "post_nms_count": post_nms_count,
+        "pre_nms_count": pre_nms_count,
+        "normalized": normalized,
+        "nms_eta": nms_eta,
+        "roi_num_type": roi_num_type
```
Comment (jiwaszki) [inline src/bindings/python/src/openvino/runtime/opset9/ops.py:191]: Is it correct to pack it into attributes? C++ implementation is giving it a specific slot in constructor. Please check it, if it somehow packs into attributes anyway, I am OK with it.

**Review by jiwaszki — CHANGES_REQUESTED** [review]
Please apply same changes to compatibility and follow-up attributes order with rest of the team.

File: src/bindings/python/src/openvino/runtime/opset9/ops.py [hunk 4]
```diff
@@ -146,3 +146,49 @@ def irdft(
...
+    :param anchors: Tensor with input anchors.
+    :param deltas: Tensor with input deltas.
+    :param scores: Tensor with input scores.
+    :param min_size: Specifies minimum box width and height.
+    :param nms_threshold: Specifies threshold to be used in the NMS stage.
+    :param pre_nms_count: Specifies number of top-n proposals before NMS.
+    :param post_nms_count: Specifies number of top-n proposals after NMS.
+    :param normalized: Specifies whether proposal bboxes are normalized or not.
+    :param nms_eta: Specifies eta parameter for adaptive NMS.
+    :param roi_num_type: Specifies element type of output roi_num.
```
[the remark this answers is not in the document]
Comment (meiyang-intel) [reply src/bindings/python/src/openvino/runtime/opset9/ops.py]: updated

File: src/bindings/python/src/openvino/runtime/opset9/ops.py [hunk 5]
```diff
@@ -146,3 +146,49 @@ def irdft(
...
+    :param im_info: Tensor with input image info.
+    :param anchors: Tensor with input anchors.
+    :param deltas: Tensor with input deltas.
+    :param scores: Tensor with input scores.
+    :param min_size: Specifies minimum box width and height.
+    :param nms_threshold: Specifies threshold to be used in the NMS stage.
+    :param pre_nms_count: Specifies number of top-n proposals before NMS.
+    :param post_nms_count: Specifies number of top-n proposals after NMS.
+    :param normalized: Specifies whether proposal bboxes are normalized or not.
+    :param nms_eta: Specifies eta parameter for adaptive NMS.
```
[the remark this answers is not in the document]
Comment (meiyang-intel) [reply src/bindings/python/src/openvino/runtime/opset9/ops.py]: updated

File: src/bindings/python/src/openvino/runtime/opset9/ops.py [hunk 6]
```diff
@@ -146,3 +146,49 @@ def irdft(
...
+
+    :param im_info: Tensor with input image info.
+    :param anchors: Tensor with input anchors.
+    :param deltas: Tensor with input deltas.
+    :param scores: Tensor with input scores.
+    :param min_size: Specifies minimum box width and height.
+    :param nms_threshold: Specifies threshold to be used in the NMS stage.
+    :param pre_nms_count: Specifies number of top-n proposals before NMS.
+    :param post_nms_count: Specifies number of top-n proposals after NMS.
+    :param normalized: Specifies whether proposal bboxes are normalized or not.
```
[the remark this answers is not in the document]
Comment (meiyang-intel) [reply src/bindings/python/src/openvino/runtime/opset9/ops.py]: updated

File: src/bindings/python/src/openvino/runtime/opset9/ops.py [hunk 7]
```diff
@@ -146,3 +146,49 @@ def irdft(
...
+    nms_eta: float = 1.0,
+    roi_num_type: str = "i64",
+    name: Optional[str] = None,
+) -> Node:
+    """Return a node which performs GenerateProposals operation.
+
+    :param im_info: Tensor with input image info.
+    :param anchors: Tensor with input anchors.
+    :param deltas: Tensor with input deltas.
+    :param scores: Tensor with input scores.
```
[the remark this answers is not in the document]
Comment (meiyang-intel) [reply src/bindings/python/src/openvino/runtime/opset9/ops.py]: updated

File: src/bindings/python/src/openvino/runtime/opset9/ops.py (hunk 2 above)
Comment (meiyang-intel) [reply src/bindings/python/src/openvino/runtime/opset9/ops.py]: updated

File: src/bindings/python/src/openvino/runtime/opset9/ops.py (hunk 3 above)
Comment (meiyang-intel) [reply src/bindings/python/src/openvino/runtime/opset9/ops.py:297]: Yes, it works. And it is checked in https://github.com/meiyang-intel/openvino/blob/c74ec58b2c947a284106141b7bb05293ac58e34c/src/bindings/python/tests/test_ngraph/test_create_op.py#L2138.

File: src/bindings/python/tests/test_ngraph/test_create_op.py [hunk 8]
```diff
@@ -2103,3 +2103,36 @@ def test_irdft():
...
+    scores_param = ov.parameter(scores_shape, name="scores")
+
+    node = ov.generate_proposals(im_info_param,
+                                 anchors_param,
+                                 deltas_param,
+                                 scores_param,
+                                 min_size=1.0,
+                                 nms_threshold=0.5,
+                                 post_nms_count=100,
+                                 pre_nms_count=200,
```
Comment (jiwaszki) [inline src/bindings/python/tests/test_ngraph/test_create_op.py:2126]:
```suggestion
                                 pre_nms_count=200,
                                 post_nms_count=100,
```
A little nit.

**Review by jiwaszki — APPROVED** [review]
LGTM now.

**Comment by meiyang-intel** [comment]
Hi @jiwaszki , if have no other concerns. Can you help merge it.

**Comment by jiwaszki** [comment]
@meiyang-intel I would like one more review and I will be happy to merge it! @p-wysocki are you able to review it?

**Pull request merged**